### PR TITLE
Type hints for `celery/schedules.py`

### PR DESCRIPTION
**Pull request by max-muoto**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryq.dev/en/main/contributing.html).

## Description

Type annotations for `celery/schedules.py` as a part of https://github.com/celery/celery/issues/7394.

Also, fixes https://github.com/celery/celery/issues/7875.

### CODE REVIEW

**Review by auvipy — CHANGES_REQUESTED** [review]
can you please check the test failures?

**Comment by max-muoto** [comment]
> can you please check the test failures?

Yep. Will debug.

File: celery/schedules.py [hunk 1]
```diff
@@ -428,14 +436,14 @@ def _expand_cronspec(cronspec, max_, min_=0):
...
-        if isinstance(cronspec, numbers.Integral):
+        if isinstance(cronspec, int):
             result = {cronspec}
         elif isinstance(cronspec, str):
             result = crontab_parser(max_, min_).parse(cronspec)
         elif isinstance(cronspec, set):
             result = cronspec
         elif isinstance(cronspec, Iterable):
-            result = set(cronspec)
+            result = set(cronspec)  # type: ignore
```
Comment (max-muoto) [inline celery/schedules.py:446]: This line was especially problematic to `mpy` for some reason, so set it as ignore. For some reason it didn't want `cronspec` to be able to be a `str` here, even though you can convert a string to a set.

File: celery/schedules.py [hunk 2]
```diff
@@ -700,8 +715,8 @@ class solar(BaseSchedule):
     Arguments:
         event (str): Solar event that triggers this task.
             See note for available values.
-        lat (int): The latitude of the observer.
-        lon (int): The longitude of the observer.
+        lat (Union[int, float]): The latitude of the observer.
```
Comment (auvipy) [inline celery/schedules.py:718]:
```suggestion
        lat (Union[float]): The latitude of the observer.
```

File: celery/schedules.py [hunk 3]
```diff
@@ -700,8 +715,8 @@ class solar(BaseSchedule):
...
+        lon (Union[int, float]): The longitude of the observer.
```
Comment (auvipy) [inline celery/schedules.py:719]:
```suggestion
        lon (Union[float]): The longitude of the observer.
```

File: celery/schedules.py [hunk 4]
```diff
@@ -665,10 +678,12 @@ def __eq__(self, other):
...
 
 
-def maybe_schedule(s, relative=False, app=None):
+def maybe_schedule(
+        s: int | float | timedelta | BaseSchedule, relative: bool = False,
+        app: Celery | None = None) -> float | timedelta | BaseSchedule:
     """Return schedule from number, timedelta, or actual schedule."""
     if s is not None:
-        if isinstance(s, numbers.Number):
+        if isinstance(s, (float, int)):
```
Comment (auvipy) [inline celery/schedules.py:686]: can you add unit test here?

**Review by auvipy — CHANGES_REQUESTED** [review]

File: celery/schedules.py (hunk 3 above)
Comment (Kaju-Bubanja) [reply celery/schedules.py:719]: You can just use float without Union

File: celery/schedules.py (hunk 2 above)
Comment (Kaju-Bubanja) [reply celery/schedules.py:718]: You can use just float no need for Union

File: celery/schedules.py (hunk 4 above)
Comment (max-muoto) [reply celery/schedules.py:686]: Will do.

File: celery/schedules.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (max-muoto) [reply celery/schedules.py]: Fixed!

Comment (max-muoto) [reply celery/schedules.py]: Fixed!

File: celery/schedules.py (hunk 3 above)
Comment (max-muoto) [reply celery/schedules.py]: Fixed

**Comment by auvipy** [comment]
I tried to fix some merge conflicts

**Comment by Nusnus** [comment]
Hey guys - is this PR ready for merge?
@max-muoto @auvipy

**Pull request merged**